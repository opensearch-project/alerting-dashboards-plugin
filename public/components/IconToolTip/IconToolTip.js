/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import PropTypes from 'prop-types';
import { EuiIcon, EuiToolTip } from '@elastic/eui';

const IconToolTip = ({ iconType, content, position = 'top' }) => (
  <EuiToolTip position={position} content={content}>
    <EuiIcon type={iconType} />
  </EuiToolTip>
);

IconToolTip.propTypes = {
  iconType: PropTypes.string.isRequired,
  content: PropTypes.string,
  position: PropTypes.string,
};

export default IconToolTip;
