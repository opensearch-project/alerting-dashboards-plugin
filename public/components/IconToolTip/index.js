/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import IconToolTip from './IconToolTip';

export default IconToolTip;
