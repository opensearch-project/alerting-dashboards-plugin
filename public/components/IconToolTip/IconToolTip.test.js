/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from 'react';
import { render } from 'enzyme';

import IconToolTip from './IconToolTip';

describe('IconToolTip', () => {
  test('renders', () => {
    const component = <IconToolTip iconType="questionInCircle" content="test content" />;
    expect(render(component)).toMatchSnapshot();
  });
});
