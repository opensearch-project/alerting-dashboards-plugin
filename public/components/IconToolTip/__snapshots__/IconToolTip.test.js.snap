// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconToolTip renders 1`] = `
<span
  class="euiToolTipAnchor"
>
  <svg
    aria-hidden="true"
    class="euiIcon euiIcon--medium euiIcon-isLoading"
    focusable="false"
    height="16"
    role="img"
    viewBox="0 0 16 16"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  />
</span>
`;
